netket/models/equivariant.py

```diff
@@ -24,54 +24,234 @@
+    precision: Any = None
+    """numerical precision of the computation see `jax.lax.Precision`for details."""
+    kernel_init: NNInitFunc = unit_normal_scaling
+    """Initializer for the Dense layer matrix."""
+    bias_init: NNInitFunc = zeros
+    """Initializer for the hidden bias."""
 
-    symmetries: Union[HashableArray, PermutationGroup]
-    """A group of symmetry operations (or array of permutation indices) over which the layer should be invariant.
+    def setup(self):
 
+        self.n_symm = np.asarray(self.symmetries).shape[0]
+
+        self.dense_symm = DenseSymmMatrix(
+            symmetries=self.symmetries,
+            features=self.features[0],
+            dtype=self.dtype,
+            use_bias=self.use_bias,
+            kernel_init=self.kernel_init,
+            bias_init=self.bias_init,
+            precision=self.precision,
+        )
+
+        self.equivariant_layers = [
+            DenseEquivariantIrrep(
+                irreps=self.irreps,
+                in_features=self.features[layer],
+                out_features=self.features[layer + 1],
+                use_bias=self.use_bias,
+                dtype=self.dtype,
+                precision=self.precision,
+                kernel_init=self.kernel_init,
+                bias_init=self.bias_init,
+            )
+            for layer in range(self.layers - 1)
+        ]
+
+    @nn.compact
+    def __call__(self, x):
+        x = self.dense_symm(x)
+
+        for layer in range(self.layers - 1):
+
+            x = self.activation(x)
+            x = self.equivariant_layers[layer](x)
+
+        if self.output_activation is not None:
+            x = self.output_activation(x)
+
+        x = x.reshape(-1, self.features[-1] * self.n_symm)
+        x_max = jnp.max(x, axis=-1, keepdims=True)
+        x = jnp.exp(x - x_max)
+        x = x.reshape(-1, self.features[-1], self.n_symm)
+        x = jnp.sum(x, 1)
+
+        x = jnp.sum(x * jnp.array(self.characters), -1)
+
+        x = jnp.log(x) + jnp.squeeze(x_max)
+
+        if self.imag_part:
+            return 1j * jnp.imag(x)
+        else:
+            return x
+
+
+class GCNN_Parity_FFT(nn.Module):
+
+    symmetries: HashableArray
+    """A group of symmetry operations (or array of permutation indices) over which the network should be equivariant.
     Numpy/Jax arrays must be wrapped into an :class:`netket.utils.HashableArray`.
     """
+    product_table: HashableArray
+    """Product table describing the algebra of the symmetry group
+    Numpy/Jax arrays must be wrapped into an :class:`netket.utils.HashableArray`.
+    """
+    shape: Tuple
+    """Shape of the translation group"""
     layers: int
     """Number of layers (not including sum layer over output)."""
-    features: Union[Tuple, int]
+    features: Tuple
     """Number of features in each layer starting from the input. If a single number is given,
     all layers will have the same number of features."""
```

Comment:
If you can give a single number than this is `Union[Tuple,int]`, otherwise change the docstring

Reply:
☝️?